addons/mail/static/src/model/model_manager.js

```diff
@@ -1075,6 +1077,23 @@ class ModelManager {
             if (!field) {
                 throw new Error(`Cannot create/update record with data unrelated to a field. (model: "${Model.modelName}", non-field attempted update: "${fieldName}")`);
             }
+
+            if (
+                field.readonly &&
+                !(options && !options.create) &&
+                !(field.compute || field.related)
+               ) {
+                throw new Error(`Can't update a readonly fields`);
+            }
+
+            if (
+                field.readonly &&
+                (field.compute || field.related) &&
+                !(options && (options.compute || options.related))
+            ) {
+                throw new Error(`Readonly related and computed can't be writed on.`);
+            }
```

Comment:
These conditions are very hard to read, but it comes from options that you give.

Instead of giving the context as option (create, compute) and let update guess what to do, it should pass a meaningful option instead that instructs update what to do (example `hasToCheckReadonly`).

Note that I think this new option should not be forwarded to `parseAndExecuteCommands`, as it was already "handled" here and makes no sense further down the chain.

With this mindset we could also pass the option to disable check when writing inverse relations (I think the main use case of read-only for us is to prevent writing manually on inverse relations when the relation is managed by a compute/related on the inverse model).